static/app/views/monitors/components/overviewTimeline/gridLines.tsx

```diff
@@ -4,31 +4,31 @@ import moment from 'moment';
 
 import DateTime from 'sentry/components/dateTime';
 import {space} from 'sentry/styles/space';
-import {TimeWindow} from 'sentry/views/monitors/components/overviewTimeline/types';
-import {
-  getStartFromTimeWindow,
-  timeWindowConfig,
-} from 'sentry/views/monitors/components/overviewTimeline/utils';
+import {TimeWindowOptions} from 'sentry/views/monitors/components/overviewTimeline/types';
 
 import {useTimelineCursor} from './timelineCursor';
 
 interface Props {
   end: Date;
-  timeWindow: TimeWindow;
+  start: Date;
+  timeWindowConfig: TimeWindowOptions;
   width: number;
   className?: string;
   showCursor?: boolean;
   stickyCursor?: boolean;
 }
 
-function clampTimeBasedOnResolution(date: moment.Moment, resolution: string) {
-  date.startOf('minute');
-  if (resolution === '1h') {
-    date.minute(date.minutes() - (date.minutes() % 10));
-  } else if (resolution === '30d') {
-    date.startOf('day');
-  } else {
+/**
+ * Aligns a date to a clean offset such as start of minute, hour, day
+ * based on the interval of how far each time label is placed.
+ */
+function alignTimeMarkersToStartOf(date: moment.Moment, timeMarkerInterval: number) {
+  if (timeMarkerInterval < 60) {
+    date.minute(date.minutes() - (date.minutes() % timeMarkerInterval));
+  } else if (timeMarkerInterval < 60 * 24) {
     date.startOf('hour');
+  } else {
+    date.startOf('day');
   }
 }
```

Comment:
Does this actually work? I thought moment was immutable so you would need to return these?

Reply:
nah this works out and seems like moments are mutable https://momentjs.com/guides/#/lib-concepts/mutability/